test/optimizer/stats_catalog_test.cpp

```diff
@@ -0,0 +1,220 @@
+#include <utility>
+
+#include "catalog/catalog.h"
+#include "catalog/postgres/pg_statistic.h"
+#include "gtest/gtest.h"
+#include "main/db_main.h"
+#include "optimizer/statistics/stats_storage.h"
+#include "parser/expression/constant_value_expression.h"
+#include "test_util/test_harness.h"
+#include "type/transient_value_factory.h"
+
+namespace terrier::optimizer {
+struct StatsCatalogTests : public TerrierTest {
+  void SetUp() override {
+    db_main_ = terrier::DBMain::Builder().SetUseGC(true).SetUseCatalog(true).Build();
+    txn_manager_ = db_main_->GetTransactionLayer()->GetTransactionManager();
+    catalog_ = db_main_->GetCatalogLayer()->GetCatalog();
+
+    auto *txn = txn_manager_->BeginTransaction();
+    db_ = catalog_->GetDatabaseOid(common::ManagedPointer(txn), catalog::DEFAULT_DATABASE);
+
+    // Ensure pg_statistic table exists with correct OID
+    auto accessor = catalog_->GetAccessor(common::ManagedPointer(txn), db_);
+    EXPECT_NE(accessor, nullptr);
```

Comment:
Why is this check required?

Reply:
It probably isn't, I just wrote this code after catalog_test which does the same check